kimchi/src/circuits/polynomials/rot.rs

```diff
@@ -0,0 +1,590 @@
+                    &table_id,
+                )
+            };
+            evals.push(combined_entry);
+        }
+
+        Evaluations::from_vec_and_domain(evals, cs.domain.d8)
+    };
+
+    // Compute the sorted plookup evaluations
+    // TODO: Once we switch to committing using lagrange commitments, `witness` will be consumed when we interpolate,
+    //       so interpolation will have to moved below this.
+    let sorted: Vec<_> = lookup::constraints::sorted(
+        dummy_lookup_value,
+        &joint_lookup_table_d8,
+        cs.domain.d1,
+        &cs.gates,
+        witness,
+        joint_combiner,
+        table_id_combiner,
+        lookup_info,
+    )
+    .map_err(|_| CircuitGateError::InvalidLookupConstraintSorted(gate_type))?;
+
+    // Randomize the last `EVALS` rows in each of the sorted polynomials in order to add zero-knowledge to the protocol.
+    let sorted: Vec<_> = sorted
+        .into_iter()
+        .map(|chunk| lookup::constraints::zk_patch(chunk, cs.domain.d1, rng))
+        .collect();
+
+    let sorted_coeffs: Vec<_> = sorted.iter().map(|e| e.clone().interpolate()).collect();
+    let sorted8 = sorted_coeffs
+        .iter()
+        .map(|v| v.evaluate_over_domain_by_ref(cs.domain.d8))
+        .collect::<Vec<_>>();
+
+    // Compute the plookup aggregation evaluations
+    let aggreg = lookup::constraints::aggregation::<_, F>(
+        dummy_lookup_value,
+        &joint_lookup_table_d8,
+        cs.domain.d1,
+        &cs.gates,
+        witness,
+        &joint_combiner,
+        &table_id_combiner,
+        *beta,
+        *gamma,
+        &sorted,
+        rng,
+        lookup_info,
+    )
+    .map_err(|_| CircuitGateError::InvalidLookupConstraintAggregation(gate_type))?;
+
+    // Precompute different forms of the aggregation polynomial for later
+    let aggreg_coeffs = aggreg.interpolate();
+    // TODO: There's probably a clever way to expand the domain without interpolating
+    let aggreg8 = aggreg_coeffs.evaluate_over_domain_by_ref(cs.domain.d8);
+
+    Ok(LookupEnvironmentData {
+        aggreg8,
+        sorted8,
+        joint_lookup_table_d8,
+    })
+}
+
+/// Get vector of rotation circuit gate types
+pub fn circuit_gates() -> [GateType; GATE_COUNT] {
+    [GateType::Rot64]
+}
+
+/// Number of constraints for a given rotation gate type
+pub fn circuit_gate_constraint_count<F: PrimeField>(typ: GateType) -> u32 {
+    match typ {
+        GateType::Rot64 => Rot64::<F>::CONSTRAINTS,
+        _ => panic!("invalid gate type"),
+    }
+}
+
+/// Get combined constraints for a given rotation circuit gate type
+pub fn circuit_gate_constraints<F: PrimeField>(typ: GateType, alphas: &Alphas<F>) -> E<F> {
+    match typ {
+        GateType::Rot64 => Rot64::combined_constraints(alphas),
+        _ => panic!("invalid gate type"),
+    }
+}
+
+/// Get the combined constraints for all rotation circuit gate types
+pub fn combined_constraints<F: PrimeField>(alphas: &Alphas<F>) -> E<F> {
```

Comment:
Nit: this seems rather redundant, given that we have `Rot64::combined_constraints` already